lib/cookie.js

```diff
@@ -869,15 +901,12 @@ CookieJar.prototype.setCookie = function(cookie, url, options, cb) {
     cookie.domain = host;
   }
 
-  // S5.3 step 7: "Otherwise, set the cookie's path to the default-path of the
-  // request-uri"
-  if (!cookie.path) {
+  //S5.2.4 If the attribute-value is empty or if the first character of the
+  //attribute-value is not %x2F ("/"):
+  //Let cookie-path be the default-path.
+  if (!cookie.path || cookie.path[0] !== '/') {
```

Comment:
This introduces another case where `.pathIsDefault` gets set. Is that correct in your opinion? Does the fall-through logic they say in S5.3 step 7 still need to happen or does the code as you have it here already cover it?